libraries/ESP8266WebServer/src/ESP8266WebServer.h

```diff
@@ -114,13 +114,15 @@ class ESP8266WebServer
   // content - actual content body
   void send(int code, const char* content_type = NULL, const String& content = String(""));
   void send(int code, char* content_type, const String& content);
+  void send(int code, const char* content_type, const char* content, size_t contentLength);
   void send(int code, const String& content_type, const String& content);
   void send_P(int code, PGM_P content_type, PGM_P content);
   void send_P(int code, PGM_P content_type, PGM_P content, size_t contentLength);
 
   void setContentLength(size_t contentLength);
   void sendHeader(const String& name, const String& value, bool first = false);
   void sendContent(const String& content);
+  void sendContent(const char* content, size_t contentLength);
```

Comment:
For many people programming in C, `const char*` implies a zero-terminated string. Of course, it can also be used to pass arbitrary binary data, but most APIs i have seen use either `const void*` or `const uint8_t*` for this purpose. I would suggest changing the type here to hint that a null-terminated string is not expected.